doc/api/buffer.markdown

```diff
@@ -21,19 +21,113 @@ The `Buffer` class is a global within Node.js, making it unlikely that one
+  the given `ArrayBuffer`.
+
+Because the behavior of `new Buffer()` changes significantly based on the type 
+of value passed as the first argument, applications that do not properly 
+validate the input arguments passed to `new Buffer()`, or that fail to 
+appropriately initialize newly allocated `Buffer` content, can inadvertently 
+introduce security and reliability issues into their code.
+
+To make the creation of `Buffer` objects more reliable and less error prone,
+the various forms of the `new Buffer()` constructor have been **deprecated**
+and replaced by separate `Buffer.from()`, `Buffer.alloc()`, and
+`Buffer.allocUnsafe()` methods. 
+
+*Developers should migrate all existing uses of the `new Buffer()` constructors 
+one of these new APIs.*
+
+* [`Buffer.from(array)`][buffer_from_array] returns a new `Buffer` containing 
+  a *copy* of the provided octets.
+* [`Buffer.from(arrayBuffer[, byteOffset [, length]])`][buffer_from_arraybuf]
+  returns a new `Buffer` that *shares* the same allocated memory as the given 
+  `ArrayBuffer`.
+* [`Buffer.from(buffer)`][buffer_from_buffer] returns a new `Buffer` 
+  containing a *copy* of the contents of the given `Buffer`.
+* [`Buffer.from(str[, encoding])`][buffer_from_string] returns a new `Buffer` 
+  containing a *copy* of the provided string.
+* [`Buffer.alloc(size[, fill[, encoding]])`][buffer_alloc] returns a "filled" 
+  `Buffer` instance of the specified size. This method can be significantly 
+  slower than [`Buffer.allocUnsafe(size)`][buffer_allocunsafe] but ensures that 
+  newly created `Buffer` instances never contain old and potentially sensitive 
+  data.
+* [`Buffer.allocUnsafe(size)`][buffer_allocunsafe] returns a new `Buffer` of 
+  the specified `size` whose content *must* be initialized using either 
+  [`buf.fill(0)`][] or written to completely.
+
+`Buffer` instances returned by `Buffer.allocUnsafe(size)` *may* be allocated
+off a shared internal memory pool if the `size` is less than or equal to half
+`Buffer.poolSize`.
+
+### The `--zero-fill-buffers` command line option
+
+Node.js can be started using the `--zero-fill-buffers` command line option to
+force all newly allocated `Buffer` and `SlowBuffer` instances created using
+either `new Buffer(size)`, `Buffer.allocUnsafe(size)`, or
+`new SlowBuffer(size)` to be *automatically zero-filled* upon creation. Use of
+this flag *changes the default behavior* of these methods and *can have a
+significant impact* on performance. Use of the `--zero-fill-buffers` option is
+recommended only when absolutely necessary to enforce that newly allocated
+`Buffer` instances cannot contain potentially sensitive data.
+
+```
+$ node --zero-fill-buffers
+> Buffer.allocUnsafe(5);
+<Buffer 00 00 00 00 00>
+```
+
+### What makes `Buffer.allocUnsafe(size)` "unsafe"?
+
+When calling `Buffer.allocUnsafe()`, the segment of allocated memory is 
+*uninitialized* (it is not zeroed-out). While this design makes the allocation 
```

Comment:
zero is just a value which we choose, right? It can be anything actually. This kinda gives me a feel that initializing means filling with zero.